.ci/ci_cache_components.sh

```diff
@@ -0,0 +1,215 @@
+
+set -o errexit
+set -o pipefail
+set -o errtrace
+
+cidir=$(dirname "$0")
+source "${cidir}/../lib/common.bash"
+source "${cidir}/lib.sh"
+
+script_name=${0##*/}
+WORKSPACE=${WORKSPACE:-$(pwd)}
+kata_dir="/usr/share/kata-containers"
+tests_repo="${tests_repo:-github.com/kata-containers/tests}"
+tests_repo_dir="${GOPATH}/src/${tests_repo}"
+
+# This builds qemu by specifying the qemu version from the
+# runtime versions file and the qemu tar file name
+cache_qemu_artifacts() {
+	pushd "${tests_repo_dir}"
+	local current_qemu_version=$(get_version "assets.hypervisor.qemu.version")
+	popd
+	local qemu_tar="kata-qemu-static.tar.gz"
+	build_cache "$qemu_tar" "${current_qemu_version}"
+}
+
+# This builds the kernel by specifying the kernel version from
+# /usr/share/kata-containers and the kernel version
+cache_kernel_artifacts() {
+	for k in "${kata_dir}"/*.container; do
+		echo "Adding ${k}"
+		local real_path=$(readlink -f "${k}")
+		#Get version from binary (format: vmlinu{z|x}-4.19.24-25)
+		local kernel_binary=$(basename "${real_path}")
+		local current_kernel_version=$(echo "${kernel_binary}" | cut -d- -f2-)
+		build_cache "${kernel_binary}" "${current_kernel_version}"
+	done
+}
+
+# This builds the image by specifying the image version from
+# /usr/share/kata-containers and the image name
+cache_image_artifacts() {
+	local image_name="image"
+	local image_path="${kata_dir}/kata-containers.img"
+	local path=$(readlink -f "${image_path}")
+	local image_version=$(echo $(basename "${path}"))
+	build_cache "${image_name}" "${image_version}"
+	create_image_tar "${image_name}"
+}
+
+# This builds nemu by specifying the nemu version from the
+# runtime versions file and the nemu tar file name
+cache_nemu_artifacts() {
+	pushd "${tests_repo_dir}"
+	local nemu_version=$(get_version "assets.hypervisor.nemu.version")
+	local nemu_tar="kata-nemu-static.tar.gz"
+	popd
+	build_cache "${nemu_tar}" "${nemu_version}"
+}
+
+# This builds the initrd image by specifying the image initrd version
+# from /usr/share/kata-containers and the image initrd name
+cache_image_initrd_artifacts() {
+	local image_name="initrd"
+	local image_path="${kata_dir}/kata-containers-initrd.img"
+	local path=$(readlink -f "${image_path}")
+	local image_version=$(echo $(basename "${path}"))
+	build_cache "${image_name}" "${image_version}"
+	create_image_tar "${image_name}"
+}
+
+# This function receives as an argument if it is an image or an image initrd
+create_image_tar(){
+	local image_name="$1"
+	local getinfo=$(cat latest-"${image_name}")
+	tar -cJf "${getinfo}.tar.xz" "${getinfo}"
+	sha256sum "${getinfo}.tar.xz" > "${image_name}-tarball.sha256sum"
+	sha256sum -c "${image_name}-tarball.sha256sum"
+	rm "${getinfo}"
+}
+
+# This function receives as arguments the component name (qemu, nemu, kernel, etc)
+# as well as the version
+build_cache() {
+	local component_name="$1"
+	local component_version="$2"
+	local check_image=$(echo "${component_version}" | grep kata || true)
```

Comment:
This is prompt to errors, as you can pass 

build_cache kemu  1.9

and wont be detected,  I wonder if something like looks better.
```
case component_name
qemu)
*)
  die invalid
```